src/common/transformations/src/transformations/common_optimizations/broadcast_transition.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/broadcast_transition.hpp"
+
+#include <memory>
+#include <openvino/opsets/opset1.hpp>
+#include <openvino/opsets/opset10.hpp>
+#include <openvino/pass/pattern/op/or.hpp>
+#include <openvino/pass/pattern/op/wrap_type.hpp>
+#include <vector>
+
+#include "itt.hpp"
+#include "transformations/utils/utils.hpp"
+
+ov::pass::BroadcastTransition::BroadcastTransition() {
+    MATCHER_SCOPE(BroadcastTransition);
+    auto bcast_data_m = pass::pattern::any_input();
+    auto target_shape_m = pass::pattern::any_input(pass::pattern::has_static_shape());
+    auto bcast_m = pass::pattern::wrap_type<opset1::Broadcast, opset10::Broadcast>({bcast_data_m, target_shape_m},
+                                                                                   pass::pattern::consumers_count(1));
+    auto eltwise_input_m = pass::pattern::any_input(pass::pattern::has_static_rank());
+    auto eltwise_1 = pass::pattern::wrap_type<op::util::BinaryElementwiseArithmetic>({eltwise_input_m, bcast_m});
+    auto eltwise_2 = pass::pattern::wrap_type<op::util::BinaryElementwiseArithmetic>({bcast_m, eltwise_input_m});
+    auto eltwise_m = std::make_shared<pass::pattern::op::Or>(OutputVector{eltwise_1, eltwise_2});
+
+    ov::matcher_pass_callback callback = [=](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        const auto eltwise = ov::as_type_ptr<ov::op::util::BinaryElementwiseArithmetic>(m.get_match_root());
+        if (eltwise->get_autob().m_type != ov::op::AutoBroadcastType::NUMPY) {
+            return false;
+        }
+
+        const auto bcast = ov::as_type_ptr<ov::opset10::Broadcast>(pattern_map.at(bcast_m).get_node_shared_ptr());
+        const auto& bcast_type = bcast->get_broadcast_spec().m_type;
+        if (bcast_type != ov::op::BroadcastType::NUMPY && bcast_type != ov::op::BroadcastType::BIDIRECTIONAL) {
+            return false;
+        }
+
+        const auto& eltwise_input = pattern_map.at(eltwise_input_m);
+        const auto& bcast_data = pattern_map.at(bcast_data_m);
+        const auto new_eltwise = eltwise->clone_with_new_inputs({eltwise_input, bcast_data});
+        ov::copy_runtime_info(eltwise, new_eltwise);
+
+        auto target_shape = pattern_map.at(target_shape_m);
+        const auto& target_shape_et = target_shape.get_element_type();
```

Comment:
Theoretically Broadcast shape et may be u32, u64 or else and ShapeOf accepts only i32 and i64 variants. However, this is quite rare -- I would say I've never seen it. As an option here we can check if et is within [i32, i64] -- then use ShapeOf's output type attribute. Otherwise, make ShapeOf i64 and insert Convert to the funky type after it.

Reply:
I corrected the code per your suggestion and added a test case. Thank you